chia/full_node/subscriptions.py

```diff
@@ -111,6 +111,54 @@ def add_coin_subscriptions(self, peer_id: bytes32, coin_ids: List[bytes32], max_
                 )
                 break
 
+    def remove_ph_subscriptions(self, peer_id: bytes32, phs: List[bytes32]) -> None:
+        puzzle_hash_peers = self._peer_puzzle_hash.get(peer_id)
+
+        # there are no subscriptions to remove
+        if puzzle_hash_peers is None:
+            return
+
+        for ph in phs:
+            if ph not in puzzle_hash_peers:
```

Comment:
you could probably save some time by removing the puzzle hash optimistically here, and if it fails, `continue`. You would avoid looking it up twice